### ISSUE TITLE
Certain markdown + front matter hangs caddy sitegen

### ISSUE THREAD

**Issue by joshix** [mholt/caddy#211]
The combination of json "front matter" and markdown shown below results in a hang with 100% cpu use when invoking caddy in a working directory containing [`bad.md`][1]. The `generated_site` directory is not created, requests are not answered.

`Caddyfile`:

```
0.0.0.0:2015
markdown / {
  ext .md
  sitegen
}
```

[`bad.md`][1]:
```markdown
{
 "title" : "CHS Maxton"
}

# *Maxton* (`maxton.local`)

Hello. My name comes from McLean's nickname for his Ideal-X, the name of
his home town, Maxton. I am the second of the *docker*(1) "container ships"
in the labs.

![the Ideal X](img/IdealX.640px.png)

* [Caddy](http://caddyserver.com)
```

Sitegen progresses normally if the front matter, or the last line before EOF, is removed.

I have reproduced this issue on linux amd64 and darwin amd64 with both the release binary 0.7.4 and with a fresh build from 36fa6e857bd22a59017ad703414c42e19a5cc1c7 by:

1. creating a directory containing the two files above
2. invoking caddy in that directory
3. observing `ps auwwx |grep caddy |grep -v grep`, `top`(1), *Activity Monitor.app*, or similar
4. observing no `generated_site` directory is created
5. observing connections refused with `curl`(1) or similar

[1]: https://gist.githubusercontent.com/joshix/cb50452916049f3c9b53/raw/812503f48dbf576373238a6d1abdb25da46be9ca/bad.md

**Comment by mholt** [comment]
I think I've traced this to some unexpected behavior with blackfriday, our Markdown library, when the renderer (which we supply) doesn't write any output for list items. Stand by...

**Comment by mholt** [comment]
Well, this was [fixed for 0.7.5](https://github.com/mholt/caddy/commit/7173764d6d2fe2c87be02db9cbe37d3368e444c8) but I forgot to mention it here. It's not ideal since I would rather not render list items as part of the summary, but that's what it'll be for now, probably until v2 of blackfriday comes out.
